Automattic/wp-calypso
# Account Recovery: transition from step 2 , step 3 to step 4

## Summary
This PR adds the transition between step 2 & 3 to step 4:
![demo-ar-1](https://cloud.githubusercontent.com/assets/1842898/24401136/b6392be4-13e5-11e7-8bf4-014c1de9cff5.gif)
![demo-ar-2](https://cloud.githubusercontent.com/assets/1842898/24401145/bccf042e-13e5-11e7-8692-8a884e9b430e.gif)

The major concern here is the way I call `page`. My previous attempt: #10965 and #12545 were both not doing well. In those two PRs, I attempted to achieve data-to-view isomorphism in a way that the URL is also taken into account. However, they both basically didn't work because they either played badly with our existent routing or screwed with the browser history. In this PR, I'm trying a much simpler approach that makes use of our existent middleware to "sort of" isolate the URL changing side effect.

## Test Plan

1. Open an incognito window and make sure you are logged out.
1. Go to http://calypso.localhost:3000/account-recovery/
1. Enter a valid, non-a11n account, and hit "continue". You should be directed to "/account-recovery/reset-password"
1. Go to http://calypso.localhost:3000/account-recovery/ again.
1. Hit "Forgot username?". You should be directed to "/account-recovery/forgot-username"
1. Enter valid personal profile of an non-a11n account, and hit "continue". You should be directed to "/account-recovery/reset-password".
